presto-main/src/main/java/com/facebook/presto/operator/SpillingUtils.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.PrestoException;
+
+import java.util.concurrent.Future;
+
+import static com.facebook.airlift.concurrent.MoreFutures.getFutureValue;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.lang.String.format;
+
+public class SpillingUtils
+{
+    private SpillingUtils() {}
+
+    public static void checkSpillSucceeded(Future spillInProgress)
+    {
+        try {
+            getFutureValue(spillInProgress);
+        }
+        catch (PrestoException prestoException) {
+            throw new PrestoException(prestoException::getErrorCode, prestoException.getMessage(), prestoException);
+        }
+        catch (RuntimeException runtimeException) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Spilling failed: %s", runtimeException.getMessage()), runtimeException);
```

Comment:
throwing `PrestoException` will still bury `runtimeException` as the cause right? -- curious why this is different from throwing the `IllegalArgumentException`?

Reply:
Yes, but it seemed an okay exception to use because it's used as a catchall for any unexpected exception we encounter. IllegalArgumentException seemed a very strange choice since the cause probably has nothing to do with illegal arguments..

Reply:
I see. So essentially what you want is something like: 
```
- Rethrow if it's already an PrestoException (so you keep all the stacktrace, etc)
-  Otherwise, thrown an wrapped exception (and may lose the stacktrace)
```

Reply:
We want both stack traces in all cases, because it's hard to know what will be helpful if you don't  know the error.  The wrapped exception is what gives us both stack traces (which is why we wrap the PrestoException in another PrestoException)